### PR TITLE
refactor(frontend): Tailwind refactor

### PR DESCRIPTION
**Pull request by adam-kov**
Moved global classes to `tailwind.config.cjs` so the [VS Code Tailwind plugin](https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss) can pick up the custom styles and offer auto completion for them. Also removed unused classes and replaced one-off elements with components to move towards a uniform appearance.

**I've checked everything I could but this may still have some breaking changes**

### CODE REVIEW

File: frontend/tailwind.config.cjs [hunk 1]
```diff
@@ -27,14 +112,130 @@ const config = {
...
+					color: theme('colors.gray.700')
+				},
+				'h1': {
+					fontSize: theme('fontSize.xl'),
+					color: theme('colors.blue.500')
+				},
+				'a': {
+					color: theme('colors.blue.500')
+				},
+				'input,input[type="text"],input[type="email"],input[type="url"],input[type="password"],input[type="number"],input[type="date"],input[type="datetime-local"],input[type="month"],input[type="search"],input[type="tel"],input[type="time"],input[type="week"],textarea,select': {
```
Comment (fatonramadani) [inline frontend/tailwind.config.cjs:142]: @adam-kov Do you know whether we can customise inputs down the line with custom TW without using important?

Comment (adam-kov) [reply frontend/tailwind.config.cjs:142]: Yes, this rule targets only by element names. Class targeting has higher specificity score, so those classes will ovveride these styles.

File: frontend/src/routes/user/login@user.svelte [hunk 2]
```diff
@@ -139,20 +145,20 @@
...
 						bind:value={password}
-						class="default-input"
+						class="mt-1"
 						id="password"
 					/>
 				</label>
-				<div class="flex flex-row-reverse  pt-4">
-					<button id="login2" class="default-button" type="button" on:click={login}> Login </button>
+				<div class="flex justify-end pt-4">
+					<Button id="login2" on:click={login}>Login</Button>
```
Comment (fatonramadani) [inline frontend/src/routes/user/login@user.svelte:161]: Do we have an ID prop on the button ?

Comment (adam-kov) [reply frontend/src/routes/user/login@user.svelte:161]: We have `{...$$restProps}`, it's newly added.

File: frontend/src/lib/components/ResourceEditor.svelte [hunk 3]
```diff
@@ -107,6 +108,8 @@
 		}
 	}
 
+	const resourceAction = () => (resourceToEdit ? editResource() : createResource())
```
Comment (rubenfiszel) [inline frontend/src/lib/components/ResourceEditor.svelte]: use fonction

File: frontend/src/lib/components/ResourceTypePicker.svelte [hunk 4]
```diff
@@ -18,48 +19,39 @@
 
 	const dispatch = createEventDispatcher()
 
-	$: {
-		if ($workspaceStore) {
-			loadResources()
-		}
+	const onClick = (resource: string | undefined) => {
```
Comment (rubenfiszel) [inline frontend/src/lib/components/ResourceTypePicker.svelte]: use function

File: frontend/src/lib/components/StringTypeNarrowing.svelte [hunk 5]
```diff
@@ -29,6 +30,15 @@
...
 	$: contentEncoding = kind == 'base64' ? 'base64' : undefined
+
+	const add = () => (enum_ = enum_ ? enum_.concat('') : [''])
+
+	const remove = (item: string) => {
+		enum_ = (enum_ || []).filter((el) => el !== item)
+		if (enum_.length == 0) {
+			enum_ = undefined
+		}
+	}
```
Comment (rubenfiszel) [inline frontend/src/lib/components/StringTypeNarrowing.svelte]: use function

**Review by rubenfiszel — APPROVED** [review]
Looks good. Merge when you have high confidence it hasn't broken anything important.

File: frontend/src/app.css [hunk 6]
```diff
@@ -1,233 +1,4 @@
...
-		@apply border rounded-md border-gray-300;
-		@apply shadow-sm;
-		@apply focus:ring focus:ring-indigo-100 focus:ring-opacity-50;
-		@apply text-sm;
-	}
-}
-
-.inline-highlight pre code.hljs {
-	padding: 0;
-	@apply text-xs;
```
Comment (rubenfiszel) [inline frontend/src/app.css]: I modified this bit in recent commits, need to port the whitespace-pre-wrap

**Review by fatonramadani — APPROVED** [review]

**Pull request merged**